src/goto-programs/goto_coverage.h

```diff
@@ -1,30 +1,71 @@
 #include <goto-programs/goto_functions.h>
 #include <goto-programs/loop_unroll.h>
 #include <langapi/language_util.h>
+#include <unordered_set>
 
 class goto_coveraget
 {
 public:
+  explicit goto_coveraget(const namespacet &ns, goto_functionst &goto_functions)
+    : ns(ns), goto_functions(goto_functions)
+  {
+    target_num = -1;
+  };
+  explicit goto_coveraget(
+    const namespacet &ns,
+    goto_functionst &goto_functions,
+    const std::string filename)
+    : ns(ns), goto_functions(goto_functions), filename(filename)
+  {
+    target_num = -1;
+  };
   // add an assert(0)
   // - at the beginning of each GOTO program
   // - at the beginning of each branch body
   // - before each END_FUNCTION statement
-  void add_false_asserts(goto_functionst &goto_functions);
+  void add_false_asserts();
 
-  // insert an assert(0) as instrumentation BEFORE each instruction
-  void
-  insert_false_assert(goto_programt &goto_program, goto_programt::targett &it);
+  void insert_assert(
+    goto_programt &goto_program,
+    goto_programt::targett &it,
+    const expr2tc &guard);
 
   // convert every assertion to an assert(0)
-  void
-  make_asserts_false(goto_functionst &goto_functions, const namespacet &ns);
+  void make_asserts_false();
 
-  int get_total_instrument() const;
+  // convert every assertion to an assert(1)
+  void make_asserts_true();
 
-  void gen_assert_instance(goto_functionst goto_functions);
+  // condition cov
+  void gen_cond_cov();
+  int get_total_instrument() const;
   int get_total_assert_instance() const;
+  std::unordered_set<std::string> get_total_cond_assert() const;
+  std::string get_filename_from_path(std::string path);
 
 protected:
-  static int total_instrument;
-  static int total_assert_instance;
+  // turn a OP b OP c into a list a, b, c
+  static void
+  collect_operands(const exprt &expr, std::list<exprt> &operands, bool &flag);
+  static void
+  collect_operators(const exprt &expr, std::list<std::string> &operators);
```

Comment:
Same, could you please document what this function is supposed to do, what kind of expressions it expects and what its result is? From the code it's not entirely clear.